common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java

```diff
@@ -569,6 +569,11 @@ protected String getBodyFromSuccessfulResponse(@NonNull final String responseBod
         return responseBody;
     }
 
+    protected String getBodyFromUnsuccessfulResponse(@NonNull final String responseBody) throws ClientException {
+        final String EMPTY_JSON_OBJECT = "{}";
+        return responseBody.isEmpty() ? EMPTY_JSON_OBJECT : responseBody;
```

Comment:
Do we know what's causing the body to be returned as empty?

Reply:
I got this empty body on a HTTP 404 error code, attempting to get the access token using and invalid Authority